ios/PolyPodApp/Core/PolyPodCore.swift

```diff
@@ -0,0 +1,51 @@
+// Please remove this line and the empty one after it
+
+import Foundation
+import PolyPodCore
+
+final class PolyPodCore {
+    static let instance = PolyPodCore()
+    let corePointer: OpaquePointer
+    
+    private let containerPath: UnsafePointer<CChar>
+    
+    init() {
+        // TODO: Handle possible errors
+        let path = try! FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: false).path
+        containerPath = NSString(string: path).utf8String!
+        
+        corePointer = new_poly_pod_core(.default(containerPath))
+    }
+}
+
+extension PolyPodCore {
+    var showInApp: Bool {
+        update_notification_show_in_app(corePointer)
+    }
+    
+    var showPush: Bool {
+        update_notification_show_push(corePointer)
+    }
+
+    func handleStartup() {
+        update_notification_handle_startup(corePointer)
+    }
+    
+    func handlePushSeen() {
+        update_notification_handle_push_seen(corePointer)
+    }
+    
+    func handleFirstRun() {
+        update_notification_handle_first_run(corePointer)
+    }
+    
+    func handleInAppSeen() {
+        update_notification_handle_in_app_seen(corePointer)
+    }
+}
+
+extension OptionKeyValueStore {
+    static func `default`(_ path: UnsafePointer<CChar>) -> OptionKeyValueStore {
+        .init(tag: .init(0), .init(.init(default_: path)))
```

Comment:
I'd be curious how it would look to provide a custom implementation of this in Swift here - maybe just put a `custom` function next to `default` that illustrates it?

Reply:
Added, https://github.com/polypoly-eu/polyPod/pull/666/commits/9fe0085e17ca3e822dbaca5506f34f54c1b97d77

Reply:
Pretty cool, thanks!